File: v2/v2_STL/implementation/PolarBear.cpp
// PolarBear.cpp

#include "PolarBear.h"
#include <iostream>
#include <thread>
#include <chrono>

using namespace std;

PolarBear::PolarBear(const Point& P) : Hewan(POLARBEAR_ID, POLARBEAR_MAXAGE)
{
	status = 1;
	power = POLARBEAR_BASEPOWER;
	deltaT = POLARBEAR_DELTAT;
	pos = P;
}

File: TubesMakhluk/src/makhluk/PolarBear.java
package makhluk;

import point.*;

/** Kelas turunan dari Hewan yang merepresentasikan seekor polar bear
 *	@author     Atika Azzahra Akbar
 *	@date       April 2016
 *	@version	1.0
 */
 
public class PolarBear extends Hewan {
	private static final char POLARBEAR_ID = 'P'; 
	private static final int POLARBEAR_MAXAGE = 250;    
	private static final int POLARBEAR_BASEPOWER = 8;
	
	/** Menciptakan objek PolarBear dengan posisi P
	 *  @param      P Point Posisi objek PolarBear
	 */
	public PolarBear(Point P){
		super(POLARBEAR_ID, POLARBEAR_MAXAGE);
		age = 0;
		status = 1;
		power = POLARBEAR_BASEPOWER;
		pos = P;
		
	}
}
